### PR TITLE
Fixes/improvements to SingleStore driver/dialect

### PR DESCRIPTION
**Pull request by mitchwadair**
Makes some changes to the driver/dialect to do:
- removes the `mode` config option and related code
  - This was only relevant to MySQL because of the `planetscale` mode
- Changes `SingleStore2` to `SingleStoreDriver`

### CODE REVIEW

File: drizzle-orm/src/singlestore-core/dialect.ts [hunk 1]
```diff
@@ -849,299 +849,4 @@ export class SingleStoreDialect {
 			selection,
 		};
 	}
-
-	buildRelationalQueryWithoutLateralSubqueries({
```
Comment (mitchwadair) [inline drizzle-orm/src/singlestore-core/dialect.ts:853]: this function was specific to the planetscale mode in the MySQL dialect

File: drizzle-orm/src/singlestore-proxy/driver.ts
Comment (Rodriguespn) [inline drizzle-orm/src/singlestore-proxy/driver.ts:1]: Andrew said to not worry about `singlestore-proxy` for now. We also deleted the IT tests for `singlestore-proxy` so I think it doesn't make sense to have this directory on this first release

I know it is not related to this PR but I wanted to bring this up @mitchwadair @palexandrefernandes

**Review by Rodriguespn — APPROVED** [review]
LGTM :+1: 

We should keep in mind that we use some of these driver classes in the integration tests so we will have to change them on rebase

**Comment by mitchwadair** [comment]
> LGTM 👍
> 
> We should keep in mind that we use some of these driver classes in the integration tests so we will have to change them on rebase

Yeah makes sense, I'll wait for that PR to merge then and I'll rebase this one

File: drizzle-orm/src/singlestore-proxy/driver.ts
Comment (mitchwadair) [reply drizzle-orm/src/singlestore-proxy/driver.ts:1]: Make sense, we can remove it in another PR I guess :+1:

Comment (Rodriguespn) [reply drizzle-orm/src/singlestore-proxy/driver.ts:1]: There is no need to wait, we can merge this into the integration tests branch right away. I rebased this branch with the integration tests branch, @mitchwadair can you confirm that all of your changes are still here after the rebase, please?

Comment (mitchwadair) [reply drizzle-orm/src/singlestore-proxy/driver.ts:1]: Yep, looks right to me

Comment (Rodriguespn) [reply drizzle-orm/src/singlestore-proxy/driver.ts:1]: Alright, just fixed the integration tests. I'm going to merge it

**Pull request merged**